posthog-ai/src/vercel/middleware.ts

```diff
@@ -49,14 +49,23 @@ const mapVercelParams = (params: any): Record<string, any> => {
 }
 
 const mapVercelPrompt = (prompt: LanguageModelV1Prompt): PostHogInput[] => {
-  return prompt.map((p) => {
+  // normalize single inputs into an array of messages
+  let promptsArray: any[]
+  if (typeof prompt === 'string') {
+    promptsArray = [{ role: 'user', content: prompt }]
+  } else if (!Array.isArray(prompt)) {
```

Comment:
just curious why we need to do this since `LanguageModelV1Prompt` is declared as `Array<LanguageModelV1Message>`.

Reply:
from testing it is not always an array